solver-next/memorycache.go

```diff
@@ -47,14 +47,78 @@ func (ck *inMemoryCacheKey) Index() Index {
 	return ck.output
 }
 
+func withExporter(ck *inMemoryCacheKey, result Result, deps [][]CacheKeyWithSelector) ExportableCacheKey {
+	return ExportableCacheKey{ck, &cacheExporter{
+		inMemoryCacheKey: ck,
+		result:           result,
+		deps:             deps,
+	}}
+}
+
+type cacheExporter struct {
+	*inMemoryCacheKey
+	result Result
+	deps   [][]CacheKeyWithSelector
+}
+
+func (ce *cacheExporter) Export(ctx context.Context, m map[digest.Digest]*ExportRecord, converter func(context.Context, Result) (*Remote, error)) (*ExportRecord, error) {
+	remote, err := converter(ctx, ce.result)
```

Comment:
Why `converter` returns a  pointer? What happens if `remote == nil && err == nil`?

Reply:
This is a valid case where the cache for the result itself is not exported but it is still present in the chain, as it may point to some other exported cache. In the current json format, this is a record with an empty digest.

Reply:
I think we should have a comment about that in the code, but it can be done later in other PRs.